src/host/ledger.h

```diff
@@ -871,16 +931,73 @@ namespace asynchost
         require_new_file = true;
       }
 
-      LOG_INFO_FMT("Setting last known/commit index to {}", idx);
       last_idx = idx;
       committed_idx = idx;
+      recovery_start_idx = recovery_start_idx_;
+
+      LOG_INFO_FMT(
+        "Set last known/commit index to {}, recovery idx to {}",
+        idx,
+        recovery_start_idx_);
+    }
+
+    void open()
+    {
+      // When the service is open, temporary recovery ledger chunks are renamed
+      // as they can now be recovered.
+      // Note: this operation cannot be rolled back.
+      LOG_INFO_FMT("Ledger open");
+
+      // Open files open for write
+      for (auto& f : files)
+      {
+        if (f->is_recovery())
+        {
+          f->open();
+        }
+      }
+
+      // Then, open committed files
+      for (auto const& f : fs::directory_iterator(ledger_dir))
```

Comment:
The files in this loop aren't necessarily seen in order of seqnos, so if something throws we may end up with a gap in the committed chunks. I don't know whether that would be handled correctly elsewhere though.

Reply:
Good point. As discussed, we now keep recovery committed files in `files` so they can be open in a stable order when the recovery is complete.